cmd/cloud_run.go

```diff
@@ -38,12 +79,87 @@ Use the "k6 cloud login" command to authenticate.`,
 			"the k6 cloud run command expects a single argument consisting in either a path to a script or "+
 				"archive file, or the \"-\" symbol indicating the script or archive should be read from stdin",
 		),
-		PreRunE: deprecatedCloudCmd.preRun,
-		RunE:    deprecatedCloudCmd.run,
+		PreRunE: cloudRunCmd.preRun,
+		RunE:    cloudRunCmd.run,
+	}
+
+	thisCmd.Flags().SortFlags = false
+	thisCmd.Flags().AddFlagSet(cloudRunCmd.flagSet())
+	thisCmd.Flags().AddFlagSet(cloudCmd.flagSet())
+
+	return thisCmd
+}
+
+func (c *cmdCloudRun) preRun(cmd *cobra.Command, args []string) error {
+	if c.localExecution {
+		if cmd.Flags().Changed("exit-on-running") {
+			return fmt.Errorf("the --local-execution flag is not compatible with the --exit-on-running flag")
+		}
+
+		if cmd.Flags().Changed("show-logs") {
+			return fmt.Errorf("the --local-execution flag is not compatible with the --show-logs flag")
+		}
+
+		return nil
+	}
+
+	if c.linger {
+		return fmt.Errorf("the --linger flag can only be used in conjunction with the --local-execution flag")
+	}
+
+	return c.deprecatedCloudCmd.preRun(cmd, args)
+}
+
+func (c *cmdCloudRun) run(cmd *cobra.Command, args []string) error {
+	if c.localExecution {
+		// Note that when running the k6 cloud run command with the --local-execution
+		// flag, we handle the no-usage-report flag here as we would do in the k6 run
+		// command.
+		return c.runCmd.run(cmd, args)
+	}
+
+	// When running the k6 cloud run command and executing in the cloud however, we
+	// explicitly disable the usage report.
```

Comment:
```suggestion
	// When running the `k6 cloud run` command explicitly disable the usage report.
```
I don't get what you mean with `and executing in the cloud however`.